src/pybind/mgr/dashboard/frontend/src/app/ceph/cephfs/cephfs-subvolume-list/cephfs-subvolume-list.component.ts

```diff
@@ -216,4 +235,20 @@ export class CephfsSubvolumeListComponent extends CdForm implements OnInit, OnCh
           })
     });
   }
+
+  selectSubVolumeGroup(subVolumeGroupName: string, index: number) {
+    this.activateButtonIndex = index;
+    this.subVolumeGroupName = subVolumeGroupName;
+    this.subVolumes$ = this.subject.pipe(
+      switchMap(() =>
+        this.cephfsSubVolume.get(this.fsName, subVolumeGroupName).pipe(
+          catchError(() => {
+            this.context.error();
+            return of(null);
+          })
+        )
+      ),
+      shareReplay(1)
+    );
```

Comment:
we have a similar async call in `ngOninit`. Let's create a function for this to remove duplication.
```
getSubVolumes(groupName = '') {
   this.subVolumes$ = this.subject.pipe(
      switchMap(() =>
        this.cephfsSubVolume.get(this.fsName, groupName).pipe(
          catchError(() => {
            this.context.error();
            return of(null);
          })
        )
      ),
      shareReplay(1)
    );
  }
```